pkgs/tools/misc/grub/pvgrub_image/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, grub2_xen }:
+
+with stdenv.lib;
+let
+  efiSystemsBuild = {
+    "i686-linux".target = "i386";
+    "x86_64-linux".target = "x86_64";
+    "aarch64-linux".target = "aarch64";
+  };
+
+in (
+
+stdenv.mkDerivation rec {
+  name = "pvgrub-image";
+
+  configs = ./configs;
+
+  buildInputs = [ grub2_xen ];
+
+  buildCommand = ''
+    cp ${configs}/* .
+    tar -cf memdisk.tar grub.cfg
+    # We include all modules except all_video.mod as otherwise grub will fail printing "no symbol table"
+    # if we include it.
+    grub-mkimage -O ${efiSystemsBuild.${stdenv.system}.target}-xen -c grub-bootstrap.cfg \
+      -m memdisk.tar -o grub-${efiSystemsBuild.${stdenv.system}.target}-xen.bin \
+      `ls ${grub2_xen}/lib/grub/${efiSystemsBuild.${stdenv.system}.target}-xen/ |grep 'mod''$'|grep -v '^all_video\.mod''$'`
```

Comment:
Actually, I had to revert back to using `ls` instead of `find`, as find produces full paths instead of just file names. If there is some specific reason not to use `ls`, please let me know. Otherwise I would prefer to stay with `ls`, as using `find` will require adding one more processing step to the pipeline.